Add xfail annotations to existing webkitpy scm_unittest

gsnedders opened the pull request:
#### 73bf03b537e5bcba6e58a3f89ca9979e98ebc3b9
<pre>
Add xfail annotations to existing webkitpy scm_unittest
<a href="https://bugs.webkit.org/show_bug.cgi?id=244292">https://bugs.webkit.org/show_bug.cgi?id=244292</a>

Reviewed by Jonathan Bedard.

* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.create_patch): Add assert to make failure more obvious
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SVNTest): Add various @xfail
(GitTest): Add various @xfail
(GitSVNTest): Add various @xfail

Canonical link: <a href="https://commits.webkit.org/257437@main">https://commits.webkit.org/257437@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6761d3dc201e4ea749db359b59eb5fb2fc34c67e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97524 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6779 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30690 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107028 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167292 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101481 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7117 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84156 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89925 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103697 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103179 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/5332 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88997 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32338 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/101093 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87205 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/20460 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75263 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/781 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21936 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/769 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/44406 "Found 1 new test failure: fast/events/blur-remove-parent-crash.html (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5122 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5582 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41312 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2019 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (approved):
Any opposition to just deleting the tests if we expect them to fail? Or is this just the first step of doing that deletion?

gsnedders committed:
Add xfail annotations to existing webkitpy scm_unittest
https://bugs.webkit.org/show_bug.cgi?id=244292

Reviewed by Jonathan Bedard.

* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.create_patch): Add assert to make failure more obvious
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SVNTest): Add various @xfail
(GitTest): Add various @xfail
(GitSVNTest): Add various @xfail

Canonical link: https://commits.webkit.org/257437@main
